frontend/src/redux/reducers/userLoginReducer.ts

```diff
@@ -0,0 +1,36 @@
+import { UserAction, UserActionTypes, UserLoginState } from 'redux/types/user';
+
+const initialState: UserLoginState = {
+  loading: false,
+  error: null,
+  isLogged: false,
+  userInfo: {}
+};
+
+export const userLoginReducer = (
+  state = initialState,
+  action: UserAction
+): UserLoginState => {
+  switch (action.type) {
+    case UserActionTypes.USER_LOGIN_REQUEST:
+      return { loading: true, error: null, isLogged: false, userInfo: {} };
+    case UserActionTypes.USER_LOGIN_SUCCESS:
+      return {
+        loading: false,
+        error: null,
+        isLogged: true,
+        userInfo: action.payload
+      };
+    case UserActionTypes.USER_LOGIN_FAIL:
+      return {
+        loading: false,
+        error: action.payload,
+        isLogged: false,
+        userInfo: {}
```

Comment:
Okay, if we are going to store user data in this reducer (userLogin), then we should call it something like userInfo, userData or similar.

Reply:
Maybe, it would be better to save only acces token in this reducer?